Automattic/jetpack
# [WIP] Carousel: Use CSS Grid for fixed info footer, add icons

This PR builds off the work in #20139, however is based off of #20107, so that we can build on top of the Swiper.JS refactor.

#### Changes proposed in this Pull Request:

* Add an apparent fixed footer to the Jetpack Carousel, with info and comment buttons
* Hide the extra info area beneath the footer by default, to prevent vertical scrolling
* When the info or comment button is clicked, reveal extra info below the fixed footer and scroll to it

#### Screenshot of WIP

![carousel-footer-with-swiper-sml](https://user-images.githubusercontent.com/14988353/123050558-319a6a00-d444-11eb-9cb5-c0e589885a44.gif)

#### To-dos

* [x] Should we hide the extra footer area when navigating back / forward through slides?
* [ ] Add smooth scrolling to scrolling up / down (note Safari iOS doesn't support this via scrollTo natively)
* [ ] Add a timeout before toggling the show class off so that we can smooth scroll up before the extra info content is hidden
* [x] Tweak the tap targets of the buttons so that they are larger — currently (with my thumbs at least) it's difficult to tap the buttons on mobile (iPhone 8)
* [ ] Tweak styling and positioning of buttons to better match designs
* [ ] Tweak styling, width and positioning of columns, info, captions, etc beneath the new footer
* [x] Update styling to ensure it works in Light mode as well as the default Dark colour scheme
* [x] Should we add in the description to the footer (as per the designs) or leave it off for the time being?
* [x] Should we add in a number indicator in the bottom left (e.g. 1/5)?
* [x] Invert the fill / background colours of the icon buttons when they're hovered.
* [x] Conditionally hide the info and comment icons based on the site's Jetpack settings for showing Exif metadata / comments:

![image](https://user-images.githubusercontent.com/14988353/123047618-dca92480-d440-11eb-8f2a-1fef2ae8e0ef.png)

#### Jetpack product discussion

Designs: p9Jlb4-2cI-p2

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

* Add an image gallery to a post on your test site
* With Jetpack active, and the Jetpack Carousel module activated, view the post on the front end of your site
* Click an image in the gallery to open up the carousel, and try out the info and comment buttons in the footer
* Go to the Jetpack settings and update the carousel to use the white colour scheme and check that the styling is correct (note: styling support for light theme hasn't been added yet) `/wp-admin/admin.php?page=jetpack#/writing`

![image](https://user-images.githubusercontent.com/14988353/123047378-9358d500-d440-11eb-8e6d-6c296ca6de6c.png)
